src/snowflake/connector/auth.py

```diff
@@ -664,3 +673,24 @@ def build_temporary_credential_name(host, user, cred_type):
     return "{host}:{user}:{driver}:{cred}".format(
         host=host.upper(), user=user.upper(), driver=KEYRING_DRIVER_NAME, cred=cred_type
     )
+
+
+def get_token_from_private_key(
+    user: str, account: str, privatekey_path: str, key_password: Optional[str]
+) -> str:
+    encoded_password = key_password.encode() if key_password is not None else None
+    with open(privatekey_path, "rb") as key:
+        p_key = load_pem_private_key(
+            key.read(), password=encoded_password, backend=default_backend()
+        )
+
+    private_key = p_key.private_bytes(
+        encoding=Encoding.DER,
+        format=PrivateFormat.PKCS8,
+        encryption_algorithm=NoEncryption(),
+    )
+    # Create new AuthByKeyPair object and set its token validity to be 24 hours (1440 min * 60)
+    auth_instance = AuthByKeyPair(private_key, 1440 * 60)
+    return auth_instance.authenticate(
```

Comment:
`authenticate` method should do the following:
```
if not '.global' in account:
    idx = account.find('.')
    if idx > 0:
        account = account[0:idx]
else:
    idx = account.find('-')
    if idx > 0:
        account = account[0:idx]
account = account.upper().replace('-', '_')
```

Reply:
Please add the test to cover all cases.

Reply:
Added this logic, and new test case has array of accounts that go through these cases

Reply:
test added in test_key_pair_authentication.py